pkg/kubectl/cmd/expose.go

```diff
@@ -162,6 +162,10 @@ func RunExpose(f *cmdutil.Factory, out io.Writer, cmd *cobra.Command, args []str
 		}
 		params["labels"] = kubectl.MakeLabels(labels)
 	}
+	if generatorName == "service/v1" {
```

Comment:
I feel like this breaks the abstraction of the generator, thus making it less useful. 

As a cleaner approach, can you move the logic inside of [`ServiceGenerator.Generate`](https://github.com/nikhiljindal/kubernetes/blob/kubectlExpose/pkg/kubectl/service.go#L47) to a package private function, then define a `ServiceGeneratorV1` and `ServiceGeneratorV2` whose `Generate` function calls the internal `generate`, setting up extra params as needed? That way clients of the Generator only pass user specified params; any defaults are handled internally according to the Generator version.